### PR TITLE
Move some /storage_proxy API endpoints to config.cc

### PR DESCRIPTION
**Pull request by xemul**
API endpoints that need a particular service to get data from are registered next to this service (#2737). In /storage_proxy function there live some endpoints that work with config, so this PR moves them to the existing config.cc with config-related endpoints. The path these endpoints are registered with remains intact, so some tweak in proxy API registration is also here.

### CODE REVIEW

**Comment by xemul** [comment]
@scylladb/scylla-maint , please review

File: api/config.cc [hunk 1]
```diff
@@ -101,10 +105,102 @@ void set_config(std::shared_ptr < api_registry_builder20 > rb, http_context& ctx
...
+    sp::get_write_rpc_timeout.unset(r);
+    sp::set_write_rpc_timeout.unset(r);
+    sp::get_counter_write_rpc_timeout.unset(r);
+    sp::set_counter_write_rpc_timeout.unset(r);
+    sp::get_cas_contention_timeout.unset(r);
+    sp::set_cas_contention_timeout.unset(r);
+    sp::get_range_rpc_timeout.unset(r);
+    sp::set_range_rpc_timeout.unset(r);
+    sp::get_truncate_rpc_timeout.unset(r);
+    sp::set_truncate_rpc_timeout.unset(r);
```
Comment (nyh) [inline api/config.cc:203]: I'm familiar with this "unset" API, but I find it unfortunate we need the developer to be diligent and remember to "unset" every route which is "set" above. A better approach would have been to save all the routes set by this object in a vector, and on unset just loop over this vector unsetting everything.

But we don't have to fix it now - it's a pre-existing approach.

**Review by nyh — APPROVED** [review]

**Comment by nyh** [comment]
There is no explicit justification for the backport/none decision. I guess that if this is long-term cleanup effort, it makes sense. But if you did these changes because it really does cause problems during shutdown (in tests? where?) then maybe backporting does make sense.

File: api/config.cc (hunk 1 above)
Comment (xemul) [reply api/config.cc:203]: @nyh , it's a long-tempting feature indeed https://github.com/scylladb/seastar/issues/1620

**Comment by xemul** [comment]
It's a long-term cleanup effort ( related to #2737 ), so no backporting

**Pull request merged**